src/lib/y2users/linux/local_reader.rb

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2users/parsers/group"
+require "y2users/parsers/passwd"
+require "y2users/parsers/shadow"
+
+module Y2Users
+  module Linux
+    # Reads local users configuration from the system using /etc files.
+    class LocalReader
+      include Yast::Logger
+
+      def initialize(source_dir = "/")
+        @source_dir = source_dir
+      end
+
+      def read_to(config)
+        config.attach(read_users + read.groups)
+        # read passwords after user, as user has to exist in advance
+        read_passwords(config)
+      end
+
+    private
+
+      attr_reader :source_dir
+
+      def read_users
+        content = File.read(File.join(source_dir, "/etc/passwd"))
+        parser = Parsers::Passwd.new
```

Comment:
NP: I would pass either `content` or the file path to the constructor. I would prefer to associate each parser object to a specific file, so `parser#parse` always returns the same result. Moreover, it will make easier to have local attributes storing e.g. partial results or whatever for the content being parsed.

Reply:
well, I try to have parser just single responsibility. Reading file and potentially handle IO exceptions for it I would like to have one level up, as e.g. getent does not have it, but File.read can have.
About having parser associated with content I also consider it, but do not see strong reasons for it and if parser is stored somewhere it can mean storing also content in memory which can be quite big on system with multiple users. That is why in the end I prefer just passing it one time.